docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,546 @@
+  Eval.later({
+    val threadId = Thread.currentThread().getId()
+    println(s"~~~>[thread: $threadId] $msg")
+    Thread.sleep(wait)
+    println(s"<~~~[thread: $threadId] $msg")
+    result
+  })
+
+def await[A](f: Future[A]): A = Await.result(f, 5 seconds)
+```
+
+We'll start implementing a data source for fetching users given their id. The identity type will be a user's
+id and the result type a user.
+
+```scala
+import fetch._
+
+type UserId = Int
+case class User(id: UserId, username: String)
+
+implicit object UserSource extends DataSource[UserId, User]{
+  override def fetch(ids: List[UserId]): Eval[Map[UserId, User]] = {
+    val result = ids.map(i => (i, User(i, s"@egg_$i"))).toMap
+    latency(result, s"Fetching users $ids")
+  }
+}
+```
+
+Now that we have a data source we can write a function for fetching users
+given an id, we just have to pass a `UserId` as an argument to `Fetch`.
+
+```scala
+def getUser(id: UserId): Fetch[User] = Fetch(id) // or, more explicitly: Fetch(id)(UserSource)
+```
+
+## Declaring and running a fetch
+
+We are now ready to declare and run fetches, the only thing missing is an instance
+of `MonadError[M, Throwable]` for our target monad (`scala.concurrent.Future`), which
+the `cats` library already provides.
+
+```scala
+import cats.std.future._
+
+val fch: Fetch[User] = getUser(1)
+val fut: Future[User] = Fetch.run(fch)
+
+await(fut)
+// ~~~>[thread: 43] Fetching users List(1)
+// <~~~[thread: 43] Fetching users List(1)
+
+//=> User(1,@egg_1)
+```
+
+In the previous example, we:
+
+- brought the implicit instance of `MonadError[Future, Throwable]` into scope importing `cats.std.future._`
+- created a fetch for a `User` using the `getUser` function
+- interpreted the fetch to a `Future` using `Fetch.run`
+
+As you can see, the fetch was executed only in one round to fetch the user and was finished after that.
+
+### Sequencing
+
+When we have two fetches that depend on each other, we can use `flatMap` to combine them. When composing fetches with `flatMap` we are telling Fetch that the second one depends on the previous one, so it isn't able to make any optimizations.
+
+```scala
+val fch: Fetch[(User, User)] = for {
+  aUser <- getUser(1)
+  anotherUser <- getUser(aUser.id + 1)
+} yield (aUser, anotherUser)
+  
+val fut: Future[(User, User)] = Fetch.run(fch)
+  
+await(fut)
+// ~~~>[thread: 44] Fetching users List(1)
+// <~~~[thread: 44] Fetching users List(1)
+// ~~~>[thread: 42] Fetching users List(2)
+// <~~~[thread: 42] Fetching users List(2)
+
+//=> (User(1,@egg_1),User(2,@egg_2))
+```
+
+### Batching
+
+If we combine two independent requests to the same data source, Fetch will
+automatically batch them together into a single request. Fetch's `join` operator
+help us tell the library that two fetches are independent, and thus can be batched
+if they use the same data source:
+
+```scala
+val fch: Fetch[(User, User)] = Fetch.join(getUser(1), getUser(2))
```

Comment:
I think the examples should use the syntax instead as it is more user friendly than the more verbose companion object methods.